pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConnectionPool.java

```diff
@@ -459,11 +470,9 @@ public void doMarkAndReleaseUselessConnections(){
             return;
         }
         List<Runnable> releaseIdleConnectionTaskList = new ArrayList<>();
-        for (Map.Entry<InetSocketAddress, ConcurrentMap<Integer, CompletableFuture<ClientCnx>>> entry :
+        for (Map.Entry<Key,  CompletableFuture<ClientCnx>> entry :
                 pool.entrySet()){
```

Comment:
Nit:

```suggestion
        for (Map.Entry<Key,  CompletableFuture<ClientCnx>> entry : pool.entrySet()) {
```

The entire block below can be indented to the left since we removed a loop.